src/App.js

```diff
@@ -1,40 +1,31 @@
-import React, { useState, useEffect } from 'react';
-import { SearchBar } from './modules/recipes/search-bar';
-import { fetchRecipesByIngredient, fetchRecipeDetailsById } from './modules/recipes/recipeService';
-import './App.css';
-import { RecipeList } from './modules/recipes/recipe-list/RecipeList';
-import { RecipeDetail } from './modules/recipes/recipe-detail/RecipeDetail.js'; 
+import { useState } from "react";
+import { UserProvider } from "./modules/user/UserContext";
+import { NavBar } from "./shared-components/NavBar";
+import { fetchRecipeDetailsById } from "./modules/recipes/recipeService";
+import { AppRouter } from "./AppRouter";
+import { useRecipes } from "./modules/recipes/RecipesProvider";
+import "./App.css";
 
 export const App = () => {
-    const [recipes, setRecipes] = useState([]);
-    const [selectedRecipe, setSelectedRecipe] = useState(null); 
-
-    useEffect(() => {
-        fetchRecipesByIngredient('chicken_breast').then((recipes) =>
-            setRecipes(recipes.length > 0 ? recipes : []),
-        );
-    }, []);
-
-    const handleRecipeClick = async (idMeal) => {
-        const details = await fetchRecipeDetailsById(idMeal);
-        setSelectedRecipe(details);
-    };
-
-    return (
-        <div className='container'>
-            <header>Recipe Search App</header>
-            <SearchBar />
-            {recipes.length > 0 ? (
-                <RecipeList recipes={recipes} onRecipeClick={handleRecipeClick} />
-            ) : (
-                <></>
-            )}
-            {selectedRecipe && <RecipeDetail recipe={selectedRecipe} />} {/* Detaylı tarif bileşenini göster */}
-        </div>
-    );
+  const [selectedRecipe, setSelectedRecipe] = useState(null);
```

Comment:
At this point, you should only use router (recipes/:id) to navigate to the recipe details page and recipes from API / context to show recipes. So remove state and `handleRecipeClick` function from here. AppRouter should not take any props.